pkg/operator/ceph/object/zone/controller.go

```diff
@@ -396,3 +425,34 @@ func (r *ReconcileObjectZone) updateStatus(observedGeneration int64, name types.
 	}
 	logger.Debugf("object zone %q status updated to %q", name, status)
 }
+
+func (r *ReconcileObjectZone) deletePoolsAndZone(objContext *object.Context, zone *cephv1.CephObjectZone, realmName string) error {
+	if !zone.Spec.PreservePoolsOnDelete {
```

Comment:
Why does preserving pools also preserve the zone? That behavior seems different than it should be to me.